sourcegraph/client.go

```diff
@@ -302,22 +302,20 @@ func addOptions(u *url.URL, opt interface{}) error {
 	return nil
 }
 
-// keepAliveTransport returns a http.RoundTripper that uses a larger
+// keepAliveTransport is an http.RoundTripper that uses a larger
 // keep-alive pool than the default.
-func keepAliveTransport() http.RoundTripper {
-	return &http.Transport{
-		Proxy: http.ProxyFromEnvironment,
-		Dial: (&net.Dialer{
-			Timeout:   30 * time.Second,
-			KeepAlive: 30 * time.Second,
-		}).Dial,
-		TLSHandshakeTimeout: 10 * time.Second,
-
-		// Allow more keep-alive connections per host to avoid
-		// ephemeral port exhaustion due to getting stuck in
-		// TIME_WAIT. Some systems have a very limited ephemeral port
-		// supply (~1024). 20 connections is perfectly reasonable,
-		// since this client will only ever hit one host.
-		MaxIdleConnsPerHost: 20,
-	}
+var keepAliveTransport http.RoundTripper = &http.Transport{
```

Comment:
I think the line `KeepAlive: 30 * time.Second` is not about HTTP keep-alives, but means TCP keep-alive packets, because it's configured on the `net` level, not on `net/http` level. When debugging, I did not see the connections being closed at all, not after 30 seconds and not later.